### DISCUSSION TITLE
Swap windows of 2 monitors

### DISCUSSION THREAD

**Discussion by ryanrhee** [rxhanson/Rectangle#1454] [Ideas]
This would be to work around a macos bug, where upon sleep and wake, all the windows that used to be on my left monitor are on my right monitor, and vice versa. (I have a 3-monitor setup: left, center, right.)

I've moved windows back to their original positions hundreds of times at this point. It would be a huge time saver if Rectangle had a way to take 2 displays and swap all the windows within those displays. Windows that span multiple displays can just be ignored, if that edge case makes this difficult.

**Comment by rxhanson** [comment]
I see no reason to turn down a good pull request for this kind of thing.

My solution to the sleep/wake rearranging lives in [Rectangle Pro](https://rectangleapp.com/pro) (paid) where you can save and apply layouts of windows (in the App layouts tab). It also has actions for moving all windows of an application to the next/prev display.

**Comment by Elijer** [comment]
This would be useful to me not because I have this same issue on sleep, but because I swap between monitors frequently, so having a hotkey that just flipped all the contents of one monitor to another would be really helpful.

Actually, the exact functionality I want is just for the most top-level programs on each monitor (when only two monitors are open) to be swapped. I can see how that's kind of niche though, interested if that is still relevant on this issue or not, and also just like, if this is already possible in the paid version, is it worth creating it for free version?

**Reply by rxhanson** [reply]
This specific functionality is not in Rectangle Pro. It has actions for moving all windows to next/previous display or one app's windows to next/previous display, or you could create custom layouts for specifying what goes where. 

I don't see myself adding this into either in the near future, but still would be willing to merge a good pull request for it in Rectangle.
